web/static/js/pages/vehicle-update.js

```diff
@@ -38,39 +84,45 @@ spa.onUnload('vehicle-update', (page) => {
    $('#vehicle-update-pic').off('change');
 });
 
-
-
 /**
  * Sends the vehicle update request to REST
  * @param  {number} id  The number of id from vehicle
  * @author Willian Conti Rezende
  */
-function updateVehicle(id){
+function updateVehicle(id, vehicle_photo_base64){
 
    // *Getting the key and the token:
    let auth = getAuthentication();
 
-
+   // *Getting data of inputs:
    let vehicle_pic = $('#vehicle-update-pic').val();
    let vehicle_title = $('#vehicle-update-title').val();
    let vehicle_manufacturer = $('#vehicle-update-manufacturer').val();
    let vehicle_type = $('#vehicle-update-type').val();
-
-
-   // *Sending a Update Vehicle to the table vehicle on data base:
-   $.ajax({
-      url: 'http://localhost:3000/api/v1/vehicles/'+id,
-      method: 'PUT',
-      headers: {'Access-Token': auth.token, 'Access-Key': auth.key},
-      data: JSON.stringify({title: vehicle_title, manufacturer: vehicle_manufacturer, type:vehicle_type, photo: vehicle_photo_base64})
-   }).done((data, textStatus, xhr) => {
-      saveAuthentication(data);
-      authenticated = true;
-      spa.navigateTo('');
-      snack.show('Vehicle updated', snack.TIME_SHORT);
+   let vehicle_year = $('#vehicle-update-year').val();
+   let vehicle_plate = $('#vehicle-update-plate').val();
+   let vehicle_revavam = $('#vehicle-update-renavam').val();
+
+
+   let data_update_vehicle = {
+      title: vehicle_title,
+      manufacturer: vehicle_manufacturer,
+      type: vehicle_type,
+      year: vehicle_year,
+      plate: vehicle_plate,
+      renavam: vehicle_revavam,
+      photo: vehicle_photo_base64
+   }
 
 
 
+   // *Sending a Update Vehicle to the table vehicle on database:
+      request.putVehicle(id, data_update_vehicle)
+         .done((data, textStatus, xhr) => {
+         saveAuthentication(data);
```

Comment:
You shouldn't mess with the authentication credentials inside of a vehicle update response handler.
Just delete this line and 123